### PR TITLE
Add Configuration-as-Code documentation and example

### PR DESCRIPTION
**Pull request by gurnben**
## Summary

As a follow-up to @Kyl-Bempah's great work on Configuration-as-Code in Konflux, I've added documentation on how to use Kustomize to manage Application and Component Configurations as Code along with a templatized/example formatted version if our CasC project!

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/configuration-as-code/proc_configuration_as_code.adoc [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+│           └── kustomization.yaml
+└── README.md
+----
+
+NOTE: You can omit `component-variant-b` if you only wish to define a single component.  This was included for illustration. 
+
+In this file structure, you will find:
+. `base`: which is a directory housing your base component yaml and a kustomize to point to the base.
+. `overlay`: which contains one or more applicatoin variants, useful when defining multiple different applications.
+. `overlay/application-a-base`: a base which defines one or more component overrides, one per component in the application and a base for your application.
```
Comment (Kyl-Bempah) [inline docs/modules/ROOT/pages/how-to-guides/configuration-as-code/proc_configuration_as_code.adoc:50]: Would this be `overlay/application-a/base`?

File: docs/modules/ROOT/pages/how-to-guides/configuration-as-code/proc_configuration_as_code.adoc [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+. `overlay/application-a-X`: An override for your application and an application specific patches for your component(s).
```
Comment (Kyl-Bempah) [inline docs/modules/ROOT/pages/how-to-guides/configuration-as-code/proc_configuration_as_code.adoc:51]: May want to change this to `overlay/application-a/<override name>` for clarity

Comment (Kyl-Bempah) [reply docs/modules/ROOT/pages/how-to-guides/configuration-as-code/proc_configuration_as_code.adoc:51]: or possibly `overlay/application-a/application-a-X`

File: docs/modules/ROOT/pages/how-to-guides/configuration-as-code/proc_configuration_as_code.adoc (hunk 1 above)
Comment (Kyl-Bempah) [reply docs/modules/ROOT/pages/how-to-guides/configuration-as-code/proc_configuration_as_code.adoc:50]: or possibly `overlay/application-a/application-a-base`

**Comment by ralphbean** [comment]
@ifireball can I ask you to review this and work with @gurnben to iterate until it's at a point where you're happy merging it?

**Review by Chr1st1anSears — APPROVED** [review]

**Pull request merged**